### PR TITLE
Emad/identity confirm

### PR DESCRIPTION
**Pull request by esdd1995**
# Description

This PR includes the following proposed change(s):

SPDBT-2978](https://jag.gov.bc.ca/jira/browse/SPDBT-2978
- refactored `CommitApplicationAsync`
- using spd_origin, defining spd_Identityconfirmed.

### CODE REVIEW

File: src/Spd.Resource.Repository/LicApp/LicAppRepository.cs [hunk 1]
```diff
@@ -24,14 +24,19 @@ public async Task<LicenceApplicationCmdResp> CommitLicenceApplicationAsync(Guid
...
+        app.statuscode = (int)Enum.Parse<ApplicationStatusOptionSet>(status.ToString());
+
+        switch (status)
         {
-            app.statuscode = (int)Enum.Parse<ApplicationStatusOptionSet>(status.ToString());
+            case ApplicationStatusEnum.Submitted:
+                app.statecode = DynamicsConstants.StateCode_Inactive;
+                break;
+            case ApplicationStatusEnum.ApplicantVerification:
+                app.spd_identityconfirmed = false;
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/LicApp/LicAppRepository.cs:35]: We should not just save the spd_identityconfirmed when user do the commit. It should be saved once the application just got created.

File: src/Spd.Manager.Licence/LicenceAppManagerBase.cs [hunk 2]
```diff
@@ -51,17 +51,26 @@ protected async Task<decimal> CommitApplicationAsync(LicenceAppBase licAppBase,
...
-            if (companionAppId != null) await _licAppRepository.CommitLicenceApplicationAsync((Guid)companionAppId, ApplicationStatusEnum.Submitted, null, ct);
-            await _licAppRepository.CommitLicenceApplicationAsync(licenceAppId, ApplicationStatusEnum.Submitted, null, ct);
+            status = isNewOrRenewal && !IsAuthenticated ? ApplicationStatusEnum.ApplicantVerification : ApplicationStatusEnum.Submitted;
         }
         else
         {
-            //companionAppId is the swl for sole proprietor which the business would pay for it, therefore the licence fee should be null here.
-            if (companionAppId != null) await _licAppRepository.CommitLicenceApplicationAsync((Guid)companionAppId, ApplicationStatusEnum.PaymentPending, null, ct);
-            await _licAppRepository.CommitLicenceApplicationAsync(licenceAppId, ApplicationStatusEnum.PaymentPending, licenceFee.Amount, ct);
+            status = isNewOrRenewal && !IsAuthenticated ? ApplicationStatusEnum.ApplicantVerification : ApplicationStatusEnum.PaymentPending;
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/LicenceAppManagerBase.cs:64]: even thought it is new and it is not authenticated, this situation should still be payment pending. Dynamics will check its status when user paid.

File: src/Spd.Resource.Repository/LicApp/LicAppRepository.cs (hunk 1 above)
Comment (peggy-quartech) [reply src/Spd.Resource.Repository/LicApp/LicAppRepository.cs:35]: the old code should not change.

**Review by peggy-quartech — CHANGES_REQUESTED** [review]

File: src/Spd.Manager.Licence/LicenceAppManagerBase.cs (hunk 2 above)
Comment (esdd1995) [reply src/Spd.Manager.Licence/LicenceAppManagerBase.cs]: changed to just be Payment Pending

File: src/Spd.Resource.Repository/LicApp/LicAppRepository.cs (hunk 1 above)
Comment (esdd1995) [reply src/Spd.Resource.Repository/LicApp/LicAppRepository.cs]: now handling spd_identityconfirmed in the repositories.

File: src/Spd.Resource.Repository/SharedRepositoryFuncs.cs [hunk 3]
```diff
@@ -63,4 +63,13 @@ public static void LinkTeam(DynamicsContext _context, string teamGuidStr, spd_ap
         return matchingAliases;
     }
 
+    public static void SetIdentityConfirmed(this spd_application app)
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/SharedRepositoryFuncs.cs:66]: move this inside the mapper

File: src/Spd.Resource.Repository/PersonLicApplication/PersonLicApplicationRepository.cs [hunk 4]
```diff
@@ -24,6 +24,7 @@ public async Task<LicenceApplicationCmdResp> CreateLicenceApplicationAsync(Creat
     {
         spd_application app = _mapper.Map<spd_application>(cmd);
         app.statuscode = (int)ApplicationStatusOptionSet.Incomplete;
+        app.SetIdentityConfirmed();
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/PersonLicApplication/PersonLicApplicationRepository.cs:27]: can we remove all these and do it inside the mapper

File: src/Spd.Resource.Repository/BizLicApplication/BizLicApplicationRepository.cs [hunk 5]
```diff
@@ -24,6 +24,7 @@ public async Task<BizLicApplicationCmdResp> CreateBizLicApplicationAsync(CreateB
         Guid applicantId;
         spd_application app = _mapper.Map<spd_application>(cmd);
         app.statuscode = (int)ApplicationStatusOptionSet.Incomplete;
+        app.SetIdentityConfirmed();
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/BizLicApplication/BizLicApplicationRepository.cs:27]: same, move inside the mapper

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs [hunk 6]
```diff
@@ -187,10 +187,11 @@ await UpdateDocumentsAsync(
     public async Task<ControllingMemberCrcAppCommandResponse> Handle(ControllingMemberCrcSubmitCommand cmd, CancellationToken ct)
     {
         ValidateFilesForNewAppAuthenticated(cmd);
+        var request = cmd.ControllingMemberCrcUpsertRequest;
         var response = await this.Handle((ControllingMemberCrcUpsertCommand)cmd, ct);
         //move files from transient bucket to main bucket when app status changed to PaymentPending.
         await MoveFilesAsync(response.ControllingMemberAppId, ct);
-        await _licAppRepository.CommitLicenceApplicationAsync(response.ControllingMemberAppId, ApplicationStatusEnum.PaymentPending, null, ct);
+        await CommitApplicationAsync(new LicenceAppBase() { ApplicationTypeCode = request.ApplicationTypeCode }, response.ControllingMemberAppId, ct);
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:194]: you need to add originalTypeCode in New LicenceAppBase

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs [hunk 7]
```diff
@@ -160,7 +160,7 @@ await ValidateInviteIdAsync(cmd.ControllingMemberCrcAppSubmitRequest.InviteId,
         await UploadNewDocsAsync(request.DocumentExpiredInfos, cmd.LicAppFileInfos, response.ControllingMemberAppId, response.ContactId, null, null, null, null, null, ct);
 
         //commit app
-        await _licAppRepository.CommitLicenceApplicationAsync(response.ControllingMemberAppId, ApplicationStatusEnum.PaymentPending, null, ct);
+        await CommitApplicationAsync(new LicenceAppBase() { ApplicationTypeCode = request.ApplicationTypeCode}, response.ControllingMemberAppId, ct);
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:163]: you need to add originalTypeCode in New LicenceAppBase

File: src/Spd.Manager.Licence/LicenceAppManagerBase.cs [hunk 8]
```diff
@@ -51,17 +51,25 @@ protected async Task<decimal> CommitApplicationAsync(LicenceAppBase licAppBase,
...
-        }
+            status = isNewOrRenewal && !IsAuthenticated ? ApplicationStatusEnum.ApplicantVerification : ApplicationStatusEnum.Submitted;
         else
+            status = ApplicationStatusEnum.PaymentPending;
+
+        // Commit the companion application if it exists
+        //companionAppId is the swl for sole proprietor which the business would pay for it, therefore the licence fee should be null here.
+        if (companionAppId != null)
         {
-            //companionAppId is the swl for sole proprietor which the business would pay for it, therefore the licence fee should be null here.
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/LicenceAppManagerBase.cs:61]: please do not remove the comments here.

File: src/Spd.Manager.Licence/LicenceAppManagerBase.cs [hunk 9]
```diff
@@ -51,17 +51,25 @@ protected async Task<decimal> CommitApplicationAsync(LicenceAppBase licAppBase,
...
-            if (companionAppId != null) await _licAppRepository.CommitLicenceApplicationAsync((Guid)companionAppId, ApplicationStatusEnum.PaymentPending, null, ct);
-            await _licAppRepository.CommitLicenceApplicationAsync(licenceAppId, ApplicationStatusEnum.PaymentPending, licenceFee.Amount, ct);
+            await _licAppRepository.CommitLicenceApplicationAsync((Guid)companionAppId, status, null, ct);
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/LicenceAppManagerBase.cs:69]: the status for companionAppId should always be payment pending.

**Review by peggy-quartech — CHANGES_REQUESTED** [review]

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**